homeassistant/components/roon/media_player.py

```diff
@@ -0,0 +1,521 @@
+                if (
+                    source["supports_standby"]
+                    and not source["status"] == "indeterminate"
+                ):
+                    self._server.roonapi.standby(self.output_id, source["control_key"])
+                    break
+        else:
+            return self.media_stop()
+
+    def set_shuffle(self, shuffle):
+        """Set shuffle state on zone."""
+        self._server.roonapi.shuffle(self.output_id, shuffle)
+
+    def select_source(self, source):
+        """Select source on player (used to sync/unsync)."""
+        _LOGGER.debug("select source called - unsync %s", self.name)
+        if source == self.name:
+            self._server.roonapi.ungroup_outputs([self.output_id])
+        else:
+            _LOGGER.debug("select source called - sync %s with %s", self.name, source)
+            output_ids = []
+            for _, zone in self._server.zones.items():
+                if zone["display_name"].lower() == source.lower():
+                    for output in zone["outputs"]:
+                        output_ids.append(output["output_id"])
+                    output_ids.append(self.output_id)
+                    self._server.roonapi.group_outputs(output_ids)
+                    break
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Send the play_media command to the media player."""
+        # Roon itself doesn't support playback of media by filename/url so this a bit of a workaround.
+        media_type = media_type.lower()
+        if media_type == "radio":
+            if self._server.roonapi.play_radio(self.zone_id, media_id):
+                self._last_playlist = media_id
+                self._last_media = media_id
+        elif media_type == "playlist":
+            if self._server.roonapi.play_playlist(
+                self.zone_id, media_id, shuffle=False
+            ):
+                self._last_playlist = media_id
+        elif media_type == "shuffleplaylist":
+            if self._server.roonapi.play_playlist(self.zone_id, media_id, shuffle=True):
+                self._last_playlist = media_id
+        elif media_type == "queueplaylist":
+            self._server.roonapi.queue_playlist(self.zone_id, media_id)
+        elif media_type == "genre":
+            self._server.roonapi.play_genre(self.zone_id, media_id)
+        elif self._server.custom_play_action:
+            # reroute the play request to the given custom script
+            _LOGGER.debug(
+                "Playback requested. Will forward to custom script/action: %s",
+                self._server.custom_play_action,
+            )
+            data = {
+                "entity_id": self.entity_id,
+                "media_type": media_type,
+                "media_id": media_id,
+            }
+            _domain, _entity = self._server.custom_play_action.split(".")
+            self.hass.services.call(_domain, _entity, data, blocking=False)
```

Comment:
We shouldn't do this. We can support a device trigger instead in a follow up PR. See eg the arcam_fmj integration.

Reply:
Again removed for now - can add in a later PR if it's important for people.